backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -139,19 +142,64 @@ public UserInfo updateUser(
     return user;
   }
 
+  /**
+   * this exists only for backward compatibility -- going forward, we should allow for a user to be
+   * updated with multiple roles at once, as in {@code updateUserPrivileges()}
+   */
+  @Deprecated
   @AuthorizationConfiguration.RequireGlobalAdminUserOrPermissionManageTargetUser
-  public OrganizationRole updateUserRole(UUID userId, OrganizationRole role) {
+  public Role updateUserRole(UUID userId, Role role) {
     ApiUser user = getApiUser(userId);
     String username = user.getLoginEmail();
     OrganizationRoleClaims orgClaims =
         _oktaRepo
             .getOrganizationRoleClaimsForUser(username)
             .orElseThrow(MisconfiguredUserException::new);
     Organization org = _orgService.getOrganization(orgClaims.getOrganizationExternalId());
-    Optional<OrganizationRoleClaims> newOrgClaims = _oktaRepo.updateUserRole(username, org, role);
-    Optional<OrganizationRole> newRole =
-        newOrgClaims.map(c -> c.getEffectiveRole().orElseThrow(MisconfiguredUserException::new));
-    return newRole.orElseThrow(MisconfiguredUserException::new);
+    Set<Facility> facilities = _orgService.getAccessibleFacilities(org, orgClaims);
+    // ensure every user altered by this method maintains all-facility access for now
+    Set<OrganizationRole> roles =
+        PermissionHolder.getEffectiveRoles(
+            EnumSet.of(
+                role.toOrganizationRole(),
+                OrganizationRole.ALL_FACILITIES,
+                OrganizationRole.getDefault()));
```

Comment:
Same comment here as in createUser.

Reply:
this is the one place right now where i think it does add value -- i'd rather not store someone as both ADMIN and ALL_FACILITIES, this allows other functions to store ADMINS as non-ALL_FACILITIES unless they explicitly add ADMINS to ALL_FACILITIES, and i don't love doing that everywhere. i don't know. i don't feel super strongly but i feel like this function generally makes role sets relatively deterministic no matter what the dev or caller tries to do and i like that.